kernel/src/vm/pmm.c

```diff
@@ -46,19 +49,39 @@ void pmm_init(size_t start, size_t end) {
     size_t infoindex = 1;
     union MemorySlice * currentslice = firstslice;
 
+    size_t peripherals_base;
 
+    switch (get_hardwareinfo()->boardType) {
+        case RaspBerryPiTwo:
+            peripherals_base = BCM2836_PERIPHERALS_PHYSICAL_BASE;
```

Comment:
Maybe make this part of the chipset interface struct. Every chipset can then define it's own base address. (Though I'd like to see what you think about this, because theoretically the same chipset implementation may be mapped on different physical addresses depending on the board)

Reply:
> Maybe make this part of the chipset interface struct. Every chipset can then define it's own base address. (Though I'd like to see what you think about this, because theoretically the same chipset implementation may be mapped on different physical addresses depending on the board)

Indeed, I think that's a good idea. I don't see the issue with the physical addresses though. The GPU, CPU and peripherals are implemented in the BCM2836 SoC. So there's no way to change the physical addresses, except by redesigning the SoC, in which case it wouldn't be a BCM2836 anymore. Though perhaps for other systems this might be the case, where the MMIO devices are external to the SoC.

Reply:
Yea, that's exactly what is was concerned about: other devices. But I think it's not really an issue as you say. This kernel is designed for pis and I doubt it will ever run on something else in its current state